lib/ranked-model/ranker.rb

```diff
@@ -210,11 +210,12 @@ def rearrange_ranks
 
       def rebalance_ranks
         if rank && instance.persisted?
-          origin = current_order.index { |item| item.instance.id == instance.id }
-          destination = current_order.index { |item| rank <= item.rank }
-          destination -= 1 if origin < destination
+          if origin = current_order.index { |item| item.instance.id == instance.id }
```

Comment:
Looks good. My only thought would be to seperate the `if` from the assignment. So assign first, then on a seperate line `if origin` just to make it easier to read.